src/transformState.js

```diff
@@ -4,8 +4,39 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+const removeKey = (action, data) => {
+  const { keysToRemove } = action;
+
+  for (const keyToRemove of keysToRemove) {
+    delete data[keyToRemove];
+  }
+};
+const clearProperties = (action, data) => {
+  for (const key in data) {
+    delete data[key];
+  }
+};
+
 function transformState(state, actions) {
-  // write code here
+  const stateCopy = state;
```

Comment:
Creating a stateCopy is unnecessary, as the original state should be modified directly. Remove this line and replace all occurrences of stateCopy with state.
You are not making a copy. You just created another link to the same object

Reply:
fixed. working with state directly now